conans/test/functional/toolchains/gnu/test_v2_autotools_template.py

```diff
@@ -167,3 +167,142 @@ def build(self):
     client2.run_command("build-release/greet")
     assert "Hello World Release!" in client2.out
 
+
+@pytest.mark.skipif(platform.system() not in ["Darwin"], reason="Only affects apple platforms")
+@pytest.mark.tool_autotools()
+def test_autotools_fix_shared_libs():
```

Comment:
This looks good!!! It's testing the right things :D 

it makes me wonder whether we want to couple the autotools toolchain _with_ the  `fix_apple_shared_install_name` function? We are now testing both things together. In the future there could be case where a new change breaks some behaviour of the autotools toolchain, but the `fix_apple_shared_install_name` function still works fine?

I think we could simplify the generation of the files maybe by doing it directly, e.g. something along the lines of:
```
echo "extern int hello;" | clang++ -x c++ -dynamiclib -o libhello.dylib  -current_version 1.2.3 -compatibility_version 1.2  -install_name /path/to/libhello.dylib - 

echo "extern int goodbye;" | clang++ -x c++ -dynamiclib -o libgoodbye.3.5.dylib -install_name /path/to/libgoodbye.3.dylib -L $(pwd) -lhello
```
In case this helps simplify the logic / reduce lines of code, while still fully testing the function :D